### PR TITLE
contrib: add qatzip compressor

### PR DESCRIPTION
**Pull request by giantcroc**
Commit Message: add qatzip compressor as contrib extension #30771 
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by soulxu** [comment]
@giantcroc would you add more description about this extension and the evidence of the user using this extension.

**Comment by giantcroc** [comment]
> @giantcroc would you like to add more description about this extension and the evidence of the user using this extension.

OK, I have added user usage in #30771. Thanks!

File: bazel/repository_locations.bzl
Comment (moderation) [inline bazel/repository_locations.bzl:1515]: Good OSSF Scorecard score 
```
scorecard --repo=https://github.com/lz4/lz4
RESULTS
-------
Aggregate score: 7.5 / 10

Check scores:
|---------|------------------------|--------------------------------|-----------------------------------------------------------------------------------------------------------------------|
|  SCORE  |          NAME          |             REASON             |                                               DOCUMENTATION/REMEDIATION                                               |
|---------|------------------------|--------------------------------|-----------------------------------------------------------------------------------------------------------------------|
| 10 / 10 | Binary-Artifacts       | no binaries found in the repo  | https://github.com/ossf/scorecard/blob/70c8e05d6dca69defac0fee1edbbf3fa8732af00/docs/checks.md#binary-artifacts       |
|---------|------------------------|--------------------------------|-----------------------------------------------------------------------------------------------------------------------|
| 1 / 10  | Branch-Protection      | branch protection is not       | https://github.com/ossf/scorecard/blob/70c8e05d6dca69defac0fee1edbbf3fa8732af00/docs/checks.md#branch-protection      |
|         |                        | maximal on development and all |                                                                                                                       |
|         |                        | release branches               |                                                                                                                       |
|---------|------------------------|--------------------------------|-----------------------------------------------------------------------------------------------------------------------|
| 10 / 10 | CI-Tests               | 16 out of 16 merged PRs        | https://github.com/ossf/scorecard/blob/70c8e05d6dca69defac0fee1edbbf3fa8732af00/docs/checks.md#ci-tests               |
|         |                        | checked by a CI test -- score  |                                                                                                                       |
|         |                        | normalized to 10               |                                                                                                                       |
|---------|------------------------|--------------------------------|-----------------------------------------------------------------------------------------------------------------------|
| 0 / 10  | CII-Best-Practices     | no effort to earn an OpenSSF   | https://github.com/ossf/scorecard/blob/70c8e05d6dca69defac0fee1edbbf3fa8732af00/docs/checks.md#cii-best-practices     |
|         |                        | best practices badge detected  |                                                                                                                       |
|---------|------------------------|--------------------------------|-----------------------------------------------------------------------------------------------------------------------|
| 8 / 10  | Code-Review            | found 2 unreviewed changesets  | https://github.com/ossf/scorecard/blob/70c8e05d6dca69defac0fee1edbbf3fa8732af00/docs/checks.md#code-review            |
|         |                        | out of 12 -- score normalized  |                                                                                                                       |
|         |                        | to 8                           |                                                                                                                       |
|---------|------------------------|--------------------------------|-----------------------------------------------------------------------------------------------------------------------|
| 10 / 10 | Contributors           | project has 52 contributing    | https://github.com/ossf/scorecard/blob/70c8e05d6dca69defac0fee1edbbf3fa8732af00/docs/checks.md#contributors           |
|         |                        | companies or organizations     |                                                                                                                       |
|---------|------------------------|--------------------------------|-----------------------------------------------------------------------------------------------------------------------|
| 10 / 10 | Dangerous-Workflow     | no dangerous workflow patterns | https://github.com/ossf/scorecard/blob/70c8e05d6dca69defac0fee1edbbf3fa8732af00/docs/checks.md#dangerous-workflow     |
|         |                        | detected                       |                                                                                                                       |
|---------|------------------------|--------------------------------|-----------------------------------------------------------------------------------------------------------------------|
| 10 / 10 | Dependency-Update-Tool | update tool detected           | https://github.com/ossf/scorecard/blob/70c8e05d6dca69defac0fee1edbbf3fa8732af00/docs/checks.md#dependency-update-tool |
|---------|------------------------|--------------------------------|-----------------------------------------------------------------------------------------------------------------------|
| 10 / 10 | Fuzzing                | project is fuzzed              | https://github.com/ossf/scorecard/blob/70c8e05d6dca69defac0fee1edbbf3fa8732af00/docs/checks.md#fuzzing                |
|---------|------------------------|--------------------------------|-----------------------------------------------------------------------------------------------------------------------|
| 9 / 10  | License                | license file detected          | https://github.com/ossf/scorecard/blob/70c8e05d6dca69defac0fee1edbbf3fa8732af00/docs/checks.md#license                |
|---------|------------------------|--------------------------------|-----------------------------------------------------------------------------------------------------------------------|
| 10 / 10 | Maintained             | 30 commit(s) out of 30 and 10  | https://github.com/ossf/scorecard/blob/70c8e05d6dca69defac0fee1edbbf3fa8732af00/docs/checks.md#maintained             |
|         |                        | issue activity out of 30 found |                                                                                                                       |
|         |                        | in the last 90 days -- score   |                                                                                                                       |
|         |                        | normalized to 10               |                                                                                                                       |
|---------|------------------------|--------------------------------|-----------------------------------------------------------------------------------------------------------------------|
| ?       | Packaging              | packaging workflow not         | https://github.com/ossf/scorecard/blob/70c8e05d6dca69defac0fee1edbbf3fa8732af00/docs/checks.md#packaging              |
|         |                        | detected                       |                                                                                                                       |
|---------|------------------------|--------------------------------|-----------------------------------------------------------------------------------------------------------------------|
| 6 / 10  | Pinned-Dependencies    | dependency not pinned by hash  | https://github.com/ossf/scorecard/blob/70c8e05d6dca69defac0fee1edbbf3fa8732af00/docs/checks.md#pinned-dependencies    |
|         |                        | detected -- score normalized   |                                                                                                                       |
|         |                        | to 6                           |                                                                                                                       |
|---------|------------------------|--------------------------------|-----------------------------------------------------------------------------------------------------------------------|
| 0 / 10  | SAST                   | SAST tool is not run on all    | https://github.com/ossf/scorecard/blob/70c8e05d6dca69defac0fee1edbbf3fa8732af00/docs/checks.md#sast                   |
|         |                        | commits -- score normalized to |                                                                                                                       |
|         |                        | 0                              |                                                                                                                       |
|---------|------------------------|--------------------------------|-----------------------------------------------------------------------------------------------------------------------|
| 10 / 10 | Security-Policy        | security policy file detected  | https://github.com/ossf/scorecard/blob/70c8e05d6dca69defac0fee1edbbf3fa8732af00/docs/checks.md#security-policy        |
|---------|------------------------|--------------------------------|-----------------------------------------------------------------------------------------------------------------------|
| 0 / 10  | Signed-Releases        | 0 out of 5 artifacts are       | https://github.com/ossf/scorecard/blob/70c8e05d6dca69defac0fee1edbbf3fa8732af00/docs/checks.md#signed-releases        |
|         |                        | signed or have provenance      |                                                                                                                       |
|---------|------------------------|--------------------------------|-----------------------------------------------------------------------------------------------------------------------|
| 10 / 10 | Token-Permissions      | GitHub workflow tokens follow  | https://github.com/ossf/scorecard/blob/70c8e05d6dca69defac0fee1edbbf3fa8732af00/docs/checks.md#token-permissions      |
|         |                        | principle of least privilege   |                                                                                                                       |
|---------|------------------------|--------------------------------|-----------------------------------------------------------------------------------------------------------------------|
| 10 / 10 | Vulnerabilities        | 0 existing vulnerabilities     | https://github.com/ossf/scorecard/blob/70c8e05d6dca69defac0fee1edbbf3fa8732af00/docs/checks.md#vulnerabilities        |
|         |                        | detected                       |                                                                                                                       |
|---------|------------------------|--------------------------------|-----------------------------------------------------------------------------------------------------------------------|
```

File: bazel/repository_locations.bzl
Comment (moderation) [inline bazel/repository_locations.bzl:453]: Over the years there has been many Intel vendor and architecture specific add-ons resulting in a large code surface area increase including some dependencies with unclear futures ie. Hyperscan (which brought in Boost as a dependency), Vectorscan. At what point would it make sense to have a separate project or repo with non-Intel specific Envoy and then adding all of the Intel specific extensions and code? The usage is these 

OSSF Scorecard

```
scorecard --repo=https://github.com/intel/qatzip
RESULTS
-------
Aggregate score: 6.2 / 10

Check scores:
|---------|------------------------|--------------------------------|-----------------------------------------------------------------------------------------------------------------------|
|  SCORE  |          NAME          |             REASON             |                                               DOCUMENTATION/REMEDIATION                                               |
|---------|------------------------|--------------------------------|-----------------------------------------------------------------------------------------------------------------------|
| 10 / 10 | Binary-Artifacts       | no binaries found in the repo  | https://github.com/ossf/scorecard/blob/70c8e05d6dca69defac0fee1edbbf3fa8732af00/docs/checks.md#binary-artifacts       |
|---------|------------------------|--------------------------------|-----------------------------------------------------------------------------------------------------------------------|
| 6 / 10  | Branch-Protection      | branch protection is not       | https://github.com/ossf/scorecard/blob/70c8e05d6dca69defac0fee1edbbf3fa8732af00/docs/checks.md#branch-protection      |
|         |                        | maximal on development and all |                                                                                                                       |
|         |                        | release branches               |                                                                                                                       |
|---------|------------------------|--------------------------------|-----------------------------------------------------------------------------------------------------------------------|
| 0 / 10  | CI-Tests               | 0 out of 4 merged PRs          | https://github.com/ossf/scorecard/blob/70c8e05d6dca69defac0fee1edbbf3fa8732af00/docs/checks.md#ci-tests               |
|         |                        | checked by a CI test -- score  |                                                                                                                       |
|         |                        | normalized to 0                |                                                                                                                       |
|---------|------------------------|--------------------------------|-----------------------------------------------------------------------------------------------------------------------|
| 0 / 10  | CII-Best-Practices     | no effort to earn an OpenSSF   | https://github.com/ossf/scorecard/blob/70c8e05d6dca69defac0fee1edbbf3fa8732af00/docs/checks.md#cii-best-practices     |
|         |                        | best practices badge detected  |                                                                                                                       |
|---------|------------------------|--------------------------------|-----------------------------------------------------------------------------------------------------------------------|
| 10 / 10 | Code-Review            | all changesets reviewed        | https://github.com/ossf/scorecard/blob/70c8e05d6dca69defac0fee1edbbf3fa8732af00/docs/checks.md#code-review            |
|---------|------------------------|--------------------------------|-----------------------------------------------------------------------------------------------------------------------|
| 6 / 10  | Contributors           | project has 2 contributing     | https://github.com/ossf/scorecard/blob/70c8e05d6dca69defac0fee1edbbf3fa8732af00/docs/checks.md#contributors           |
|         |                        | companies or organizations --  |                                                                                                                       |
|         |                        | score normalized to 6          |                                                                                                                       |
|---------|------------------------|--------------------------------|-----------------------------------------------------------------------------------------------------------------------|
| ?       | Dangerous-Workflow     | no workflows found             | https://github.com/ossf/scorecard/blob/70c8e05d6dca69defac0fee1edbbf3fa8732af00/docs/checks.md#dangerous-workflow     |
|---------|------------------------|--------------------------------|-----------------------------------------------------------------------------------------------------------------------|
| 10 / 10 | Dependency-Update-Tool | update tool detected           | https://github.com/ossf/scorecard/blob/70c8e05d6dca69defac0fee1edbbf3fa8732af00/docs/checks.md#dependency-update-tool |
|---------|------------------------|--------------------------------|-----------------------------------------------------------------------------------------------------------------------|
| 0 / 10  | Fuzzing                | project is not fuzzed          | https://github.com/ossf/scorecard/blob/70c8e05d6dca69defac0fee1edbbf3fa8732af00/docs/checks.md#fuzzing                |
|---------|------------------------|--------------------------------|-----------------------------------------------------------------------------------------------------------------------|
| 9 / 10  | License                | license file detected          | https://github.com/ossf/scorecard/blob/70c8e05d6dca69defac0fee1edbbf3fa8732af00/docs/checks.md#license                |
|---------|------------------------|--------------------------------|-----------------------------------------------------------------------------------------------------------------------|
| 0 / 10  | Maintained             | 0 commit(s) out of 30 and 0    | https://github.com/ossf/scorecard/blob/70c8e05d6dca69defac0fee1edbbf3fa8732af00/docs/checks.md#maintained             |
|         |                        | issue activity out of 30 found |                                                                                                                       |
|         |                        | in the last 90 days -- score   |                                                                                                                       |
|         |                        | normalized to 0                |                                                                                                                       |
|---------|------------------------|--------------------------------|-----------------------------------------------------------------------------------------------------------------------|
| ?       | Packaging              | packaging workflow not         | https://github.com/ossf/scorecard/blob/70c8e05d6dca69defac0fee1edbbf3fa8732af00/docs/checks.md#packaging              |
|         |                        | detected                       |                                                                                                                       |
|---------|------------------------|--------------------------------|-----------------------------------------------------------------------------------------------------------------------|
| ?       | Pinned-Dependencies    | no dependencies found          | https://github.com/ossf/scorecard/blob/70c8e05d6dca69defac0fee1edbbf3fa8732af00/docs/checks.md#pinned-dependencies    |
|---------|------------------------|--------------------------------|-----------------------------------------------------------------------------------------------------------------------|
| 0 / 10  | SAST                   | SAST tool is not run on all    | https://github.com/ossf/scorecard/blob/70c8e05d6dca69defac0fee1edbbf3fa8732af00/docs/checks.md#sast                   |
|         |                        | commits -- score normalized to |                                                                                                                       |
|         |                        | 0                              |                                                                                                                       |
|---------|------------------------|--------------------------------|-----------------------------------------------------------------------------------------------------------------------|
| 10 / 10 | Security-Policy        | security policy file detected  | https://github.com/ossf/scorecard/blob/70c8e05d6dca69defac0fee1edbbf3fa8732af00/docs/checks.md#security-policy        |
|---------|------------------------|--------------------------------|-----------------------------------------------------------------------------------------------------------------------|
| ?       | Signed-Releases        | no releases found              | https://github.com/ossf/scorecard/blob/70c8e05d6dca69defac0fee1edbbf3fa8732af00/docs/checks.md#signed-releases        |
|---------|------------------------|--------------------------------|-----------------------------------------------------------------------------------------------------------------------|
| ?       | Token-Permissions      | no github tokens found         | https://github.com/ossf/scorecard/blob/70c8e05d6dca69defac0fee1edbbf3fa8732af00/docs/checks.md#token-permissions      |
|---------|------------------------|--------------------------------|-----------------------------------------------------------------------------------------------------------------------|
```

**Review by moderation — COMMENTED** [review]
Time for an Intel specific project to slim down vendor and platform specific features in core Envoy?

**Comment by soulxu** [comment]
> Time for an Intel specific project to slim down vendor and platform specific features in core Envoy?

I talked with @giantcroc. Today we get customer adoption, which is good news for us, and then we want to see if we reach the requirement to contribute it to the upstream. 

From my understanding, the purpose of contributing to the open-source project is that it is hard for us to reach more customers, the upstream project helps us to get wider adoption and feedback.  Just like this one https://github.com/envoyproxy/envoy/issues/21531#issuecomment-1790551233

Also, contribute open-source project, and we get some reviews and feedback from the community, that help us to integrate our stuff correctly with the upstream project.

That is all we appreciate what we get from the community.

We know it will give a lot of burden to the community, that is why we submit to the contrib extension not the Envoy core (accept anything can help on common use-case)

But yes, it is also hard for us to change the whole intel to match the upstream requirement, but just try our best. It is also fine if we are really far from the least requirement, we accept that, and hope we can come back in the future when we get better.

**Comment by giantcroc** [comment]
> > Time for an Intel specific project to slim down vendor and platform specific features in core Envoy?
> 
> I talked with @giantcroc. Today we get customer adoption, which is good news for us, and then we want to see if we reach the requirement to contribute it to the upstream.
> 
> From my understanding, the purpose of contributing to the open-source project is that it is hard for us to reach more customers, the upstream project helps us to get wider adoption and feedback. Just like this one [#21531 (comment)](https://github.com/envoyproxy/envoy/issues/21531#issuecomment-1790551233)
> 
> Also, contribute open-source project, and we get some reviews and feedback from the community, that help us to integrate our stuff correctly with the upstream project.
> 
> That is all we appreciate what we get from the community.
> 
> We know it will give a lot of burden to the community, that is why we submit to the contrib extension not the Envoy core (except anything can benefit for common use-case)
> 
> But yes, it is also hard for us to change the whole intel to match the upstream requirement, but just try our best. It is also fine if we are really far from the least requirement, we accept that, and hope we can come back in the future when we get better.

Yeah, agree with @soulxu, we are very glad to have customer's usage case now,  and we really hope this could meet  community's requirements, if not, we are prepared to do better.

**Comment by giantcroc** [comment]
@mattklein123 Hi Matt, do you think there is an opportunity for us to contribute qatzip compressor to the community? And it's fine if we don't match the requirement, we will try in the future.

**Comment by yanavlasov** [comment]
Following up with maintainers

**Comment by yanavlasov** [comment]
Please resolve Windows build error before I can merge it.

/wait

File: contrib/qat/compression/qatzip/compressor/test/BUILD
Comment (soulxu) [inline contrib/qat/compression/qatzip/compressor/test/BUILD:13]: Probably add this to skip test on arm

``target_compatible_with = envoy_contrib_linux_x86_64_constraints(),``

Comment (giantcroc) [reply contrib/qat/compression/qatzip/compressor/test/BUILD]: Yeah, I have added the compile option like the test of dlb, it should work.

**Review by yanavlasov — APPROVED** [review]

**Comment by brunsgaard** [comment]
If this PR is merged I have issues compiling on MacOS. Mainly `bazel build @envoy//bazel/foreign_cc:lz4`

```
bazel build @envoy//bazel/foreign_cc:lz4 --verbose_failures
INFO: Analyzed target //bazel/foreign_cc:lz4 (0 packages loaded, 0 targets configured).
INFO: Found 1 target...
ERROR: /Users/jonas.brunsgaard/code/strimzi-wip/envoy2/envoy/bazel/foreign_cc/BUILD:63:5: Foreign Cc - Make: Building lz4 failed: (Exit 2): bash failed: error executing command (from target //bazel/foreign_cc:lz4)
  (cd /private/var/tmp/_bazel_jonas.brunsgaard/b87ffa8d691acaee9dfa9a389da51f08/sandbox/darwin-sandbox/19226/execroot/envoy && \
  exec env - \
    PATH=/opt/homebrew/bin:/opt/local/bin:/usr/local/bin:/usr/bin:/bin \
  /bin/bash -c bazel-out/darwin-fastbuild/bin/bazel/foreign_cc/lz4_foreign_cc/wrapper_build_script.sh)
# Configuration: 293240611eb29bd26a1d18e2f6bc646c3721901cd716a4ec57297f597907ae3e
# Execution platform: @local_config_platform//:host

Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
rules_foreign_cc: Build failed!
rules_foreign_cc: Keeping temp build directory and dependencies directory for debug.
rules_foreign_cc: Please note that the directories inside a sandbox are still cleaned unless you specify --sandbox_debug Bazel command line flag.
rules_foreign_cc: Printing build logs:
_____ BEGIN BUILD LOGS _____

Bazel external C/C++ Rules. Building library lz4

Environment:______________
XCODE_VERSION_OVERRIDE=15.1.0.15C65
BUILD_SCRIPT=bazel-out/darwin-fastbuild/bin/bazel/foreign_cc/lz4_foreign_cc/build_script.sh
EXT_BUILD_ROOT=/private/var/tmp/_bazel_jonas.brunsgaard/b87ffa8d691acaee9dfa9a389da51f08/sandbox/darwin-sandbox/19226/execroot/envoy
BUILD_LOG=bazel-out/darwin-fastbuild/bin/bazel/foreign_cc/lz4_foreign_cc/Make.log
PWD=/private/var/tmp/_bazel_jonas.brunsgaard/b87ffa8d691acaee9dfa9a389da51f08/sandbox/darwin-sandbox/19226/execroot/envoy
SDKROOT=/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk
BUILD_WRAPPER_SCRIPT=bazel-out/darwin-fastbuild/bin/bazel/foreign_cc/lz4_foreign_cc/wrapper_build_script.sh
TMPDIR=/var/folders/yp/sg57x5597798plt6918hpzjr0000gn/T/
CMAKE_OSX_ARCHITECTURES=x86_64
EXT_BUILD_DEPS=/private/var/tmp/_bazel_jonas.brunsgaard/b87ffa8d691acaee9dfa9a389da51f08/sandbox/darwin-sandbox/19226/execroot/envoy/bazel-out/darwin-fastbuild/bin/bazel/foreign_cc/lz4.ext_build_deps
APPLE_SDK_PLATFORM=MacOSX
APPLE_SDK_VERSION_OVERRIDE=14.2
BUILD_TMPDIR=/private/var/tmp/_bazel_jonas.brunsgaard/b87ffa8d691acaee9dfa9a389da51f08/sandbox/darwin-sandbox/19226/execroot/envoy/bazel-out/darwin-fastbuild/bin/bazel/foreign_cc/lz4.build_tmpdir
SHLVL=3
ZERO_AR_DATE=1
INSTALLDIR=/private/var/tmp/_bazel_jonas.brunsgaard/b87ffa8d691acaee9dfa9a389da51f08/sandbox/darwin-sandbox/19226/execroot/envoy/bazel-out/darwin-fastbuild/bin/bazel/foreign_cc/lz4
PATH=/private/var/tmp/_bazel_jonas.brunsgaard/b87ffa8d691acaee9dfa9a389da51f08/sandbox/darwin-sandbox/19226/execroot/envoy:/opt/homebrew/bin:/opt/local/bin:/usr/local/bin:/usr/bin:/bin
DEVELOPER_DIR=/Library/Developer/CommandLineTools
__CF_USER_TEXT_ENCODING=0x1F5:0x0:0x0
_=/usr/bin/env
__________________________
+ ASFLAGS='-D_FORTIFY_SOURCE=1 -fstack-protector -fcolor-diagnostics -Wall -Wthread-safety -Wself-assign -fno-omit-frame-pointer -O0 -DDEBUG DEBUG_PREFIX_MAP_PWD=. -isysroot __BAZEL_XCODE_SDKROOT__ -F__BAZEL_XCODE_SDKROOT__/System/Library/Frameworks -F__BAZEL_XCODE_DEVELOPER_DIR__/Platforms/MacOSX.platform/Developer/Library/Frameworks -no-canonical-prefixes -pthread -no-canonical-prefixes -Wno-builtin-macro-redefined -D__DATE__=redacted -D__TIMESTAMP__=redacted -D__TIME__=redacted -target x86_64-apple-macosx14.2 -DABSL_MIN_LOG_LEVEL=4'
+ CFLAGS='-D_FORTIFY_SOURCE=1 -fstack-protector -fcolor-diagnostics -Wall -Wthread-safety -Wself-assign -fno-omit-frame-pointer -O0 -DDEBUG DEBUG_PREFIX_MAP_PWD=. -isysroot __BAZEL_XCODE_SDKROOT__ -F__BAZEL_XCODE_SDKROOT__/System/Library/Frameworks -F__BAZEL_XCODE_DEVELOPER_DIR__/Platforms/MacOSX.platform/Developer/Library/Frameworks -no-canonical-prefixes -pthread -no-canonical-prefixes -Wno-builtin-macro-redefined -D__DATE__=redacted -D__TIMESTAMP__=redacted -D__TIME__=redacted -target x86_64-apple-macosx14.2 -DABSL_MIN_LOG_LEVEL=4'
+ CXXFLAGS='-D_FORTIFY_SOURCE=1 -fstack-protector -fcolor-diagnostics -Wall -Wthread-safety -Wself-assign -fno-omit-frame-pointer -O0 -DDEBUG -std=c++11 DEBUG_PREFIX_MAP_PWD=. -isysroot __BAZEL_XCODE_SDKROOT__ -F__BAZEL_XCODE_SDKROOT__/System/Library/Frameworks -F__BAZEL_XCODE_DEVELOPER_DIR__/Platforms/MacOSX.platform/Developer/Library/Frameworks -no-canonical-prefixes -pthread -no-canonical-prefixes -Wno-builtin-macro-redefined -D__DATE__=redacted -D__TIMESTAMP__=redacted -D__TIME__=redacted -target x86_64-apple-macosx14.2 -DABSL_MIN_LOG_LEVEL=4 -std=c++17'
+ LDFLAGS='-lc++ -fobjc-link-runtime -headerpad_max_install_names -no-canonical-prefixes -target x86_64-apple-macosx14.2 -Xlinker -no_deduplicate -lc++ -target x86_64-apple-macosx14.2'
+ AR=/private/var/tmp/_bazel_jonas.brunsgaard/b87ffa8d691acaee9dfa9a389da51f08/sandbox/darwin-sandbox/19226/execroot/envoy/external/local_config_cc/libtool
+ CC=/private/var/tmp/_bazel_jonas.brunsgaard/b87ffa8d691acaee9dfa9a389da51f08/sandbox/darwin-sandbox/19226/execroot/envoy/external/local_config_cc/wrapped_clang
+ CXX=/private/var/tmp/_bazel_jonas.brunsgaard/b87ffa8d691acaee9dfa9a389da51f08/sandbox/darwin-sandbox/19226/execroot/envoy/external/local_config_cc/wrapped_clang_pp
+ RANLIB=:
+ CPPFLAGS=
+ /private/var/tmp/_bazel_jonas.brunsgaard/b87ffa8d691acaee9dfa9a389da51f08/sandbox/darwin-sandbox/19226/execroot/envoy/bazel-out/darwin-opt-exec-2B5CBBC6/bin/external/rules_foreign_cc/toolchains/make/bin/make lib MOREFLAGS=-fPIC BUILD_SHARED=no PREFIX=/private/var/tmp/_bazel_jonas.brunsgaard/b87ffa8d691acaee9dfa9a389da51f08/sandbox/darwin-sandbox/19226/execroot/envoy/bazel-out/darwin-fastbuild/bin/bazel/foreign_cc/lz4
make[1]: Entering directory '/private/var/tmp/_bazel_jonas.brunsgaard/b87ffa8d691acaee9dfa9a389da51f08/sandbox/darwin-sandbox/19226/execroot/envoy/bazel-out/darwin-fastbuild/bin/bazel/foreign_cc/lz4.build_tmpdir/lib'
compiling static library
error: /Library/Developer/CommandLineTools/usr/bin/libtool: no output file specified (specify with -o output)
Usage: /Library/Developer/CommandLineTools/usr/bin/libtool -static [-] file [...] [-filelist listfile[,dirname]] [-arch_only arch] [-sacLT] [-no_warning_for_no_symbols]
Usage: /Library/Developer/CommandLineTools/usr/bin/libtool -dynamic [-] file [...] [-filelist listfile[,dirname]] [-arch_only arch] [-o output] [-install_name name] [-compatibility_version #] [-current_version #] [-seg1addr 0x#] [-segs_read_only_addr 0x#] [-segs_read_write_addr 0x#] [-seg_addr_table <filename>] [-seg_addr_table_filename <file_system_path>] [-all_load] [-noall_load]
make[1]: *** [Makefile:99: liblz4.a] Error 1
make[1]: Leaving directory '/private/var/tmp/_bazel_jonas.brunsgaard/b87ffa8d691acaee9dfa9a389da51f08/sandbox/darwin-sandbox/19226/execroot/envoy/bazel-out/darwin-fastbuild/bin/bazel/foreign_cc/lz4.build_tmpdir/lib'
make: *** [Makefile:57: liblz4.a] Error 2
_____ END BUILD LOGS _____
rules_foreign_cc: Build wrapper script location: bazel-out/darwin-fastbuild/bin/bazel/foreign_cc/lz4_foreign_cc/wrapper_build_script.sh
rules_foreign_cc: Build script location: bazel-out/darwin-fastbuild/bin/bazel/foreign_cc/lz4_foreign_cc/build_script.sh
rules_foreign_cc: Build log location: bazel-out/darwin-fastbuild/bin/bazel/foreign_cc/lz4_foreign_cc/Make.log

Target //bazel/foreign_cc:lz4 failed to build
INFO: Elapsed time: 4.034s, Critical Path: 3.37s
INFO: 2 processes: 2 internal.
FAILED: Build did NOT complete successfully
```

**Pull request merged**
